### PR TITLE
data bundles for iOS

### PR DESCRIPTION
**Pull request by rohandandavati**
Add data bundles to iOS - tests for response fetch

### CODE REVIEW

File: Firebase/InAppMessaging/Data/FIRIAMFetchResponseParser.m [hunk 1]
```diff
@@ -336,13 +336,20 @@ - (FIRIAMMessageDefinition *)convertToMessageDefinitionWithMessageDict:(NSDictio
                                                contentData:msgData
                                            renderingEffect:renderEffect];
 
+    /// Is this necessary, or can I just do NSDictionary * appData = messageNode[@"dataBundle"]?
```
Comment (christibbs) [inline Firebase/InAppMessaging/Data/FIRIAMFetchResponseParser.m]: Yes, keep this in. In SDK code there's a premium on safety.

File: Firebase/InAppMessaging/Data/FIRIAMMessageDefinition.h [hunk 2]
```diff
@@ -43,8 +46,13 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)initWithRenderData:(FIRIAMMessageRenderData *)renderData
                          startTime:(NSTimeInterval)startTime
                            endTime:(NSTimeInterval)endTime
-                 triggerDefinition:(NSArray<FIRIAMDisplayTriggerDefinition *> *)renderTriggers;
+                 triggerDefinition:(NSArray<FIRIAMDisplayTriggerDefinition *> *)renderTriggers
+                           appData:(nullable NSDictionary *)appData;
```
Comment (christibbs) [inline Firebase/InAppMessaging/Data/FIRIAMMessageDefinition.h]: Add `NS_DESIGNATED_INITIALIZER` here (see `FIRIAMActionURLFollower.h` line 36).

**Comment by christibbs** [comment]
Fix the duplicate semi-colon after `NS_DESIGNATED_INITIALIZER` in the message definition class. Also, run `scripts/style.sh`-- that's what's causing the build failure.

**Comment by rohandandavati** [comment]
Ah yeah, thanks for catching that

**Comment by christibbs** [comment]
Hold off on merging this until we figure out a merge strategy with the consolidation stuff.

**Comment by paulb777** [comment]
@rohandandavati Can this PR be closed?  It looks like it was redone separately.

**Pull request closed without merging**